src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -20424,6 +20424,1466 @@ public void AppendFormatted<T>(T hole, int alignment = 0, string format = null)
+            // (1,18): error CS8949: The InterpolatedStringHandlerArgumentAttribute applied to parameter 'InterpolationHandler h' is malformed and cannot be interpreted. Construct an instance of 'InterpolationHandler' manually.
+            // 1.InstanceMethod($"");
+            Diagnostic(ErrorCode.ERR_InterpolatedStringHandlerArgumentAttributeMalformed, @"$""""").WithArguments("InterpolationHandler h", "InterpolationHandler").WithLocation(1, 18),
+            // (1,18): error CS7036: There is no argument given that corresponds to the required parameter 'param' of 'InterpolationHandler.InterpolationHandler(int, int, int)'
+            // 1.InstanceMethod($"");
+            Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, @"$""""").WithArguments("param", "InterpolationHandler.InterpolationHandler(int, int, int)").WithLocation(1, 18),
+            // (1,18): error CS1615: Argument 3 may not be passed with the 'out' keyword
+            // 1.InstanceMethod($"");
+            Diagnostic(ErrorCode.ERR_BadArgExtraRef, @"$""""").WithArguments("3", "out").WithLocation(1, 18),
+            // (2,21): error CS8949: The InterpolatedStringHandlerArgumentAttribute applied to parameter 'InterpolationHandler h' is malformed and cannot be interpreted. Construct an instance of 'InterpolationHandler' manually.
+            // E.InstanceMethod(1, $"");
+            Diagnostic(ErrorCode.ERR_InterpolatedStringHandlerArgumentAttributeMalformed, @"$""""").WithArguments("InterpolationHandler h", "InterpolationHandler").WithLocation(2, 21),
+            // (2,21): error CS7036: There is no argument given that corresponds to the required parameter 'param' of 'InterpolationHandler.InterpolationHandler(int, int, int)'
+            // E.InstanceMethod(1, $"");
+            Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, @"$""""").WithArguments("param", "InterpolationHandler.InterpolationHandler(int, int, int)").WithLocation(2, 21),
+            // (2,21): error CS1615: Argument 3 may not be passed with the 'out' keyword
+            // E.InstanceMethod(1, $"");
+            Diagnostic(ErrorCode.ERR_BadArgExtraRef, @"$""""").WithArguments("3", "out").WithLocation(2, 21)
+        );
+    }
+
+    [Fact]
+    public void InterpolationHandler_Indexer_InObjectInitializer()
+    {
+        var code = """
+            _ = new C() { [$"{1}"] = 1 };
+
+            public class C
+            {
+            }
+
+            [System.Runtime.CompilerServices.InterpolatedStringHandler]
+            public struct CustomHandler
+            {
+                public CustomHandler(int literalLength, int formattedCount, C c)
+                {
+                }
+
+                public void AppendFormatted(int i) {}
+            }
+
+            public static class CExt
+            {
+                extension(C c)
+                {
+                    public int this[[System.Runtime.CompilerServices.InterpolatedStringHandlerArgument("c")] CustomHandler h]
```

Comment:
>InterpolatedStringHandlerArgument [](http://example.com/codeflow?start=69&length=33)

Do we have tests covering application of this attribute to receiver parameter itself?